### PR TITLE
Add support for samplers to modify Tracestate

### LINKED ISSUE #1708: Samplers should be allowed to modify tracestate
https://github.com/open-telemetry/opentelemetry-specification/pull/988 Spec allows samplers to modify trace state and have it reflected in the created Span.
This requires support from .NET runtime, so need to brought up to .NET repo.  Opening here to visibility and to see if any workarounds can be used until then.

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #1708
To maintain backward compatibility, `SamplingResult` will default to `null` for `TraceStateString`. Any non-null value set by samplers, will be set as the tracestate of the created activity.
If sampler intents to not modify tracestate, it can return null. (Default)
Else, it can populate tracestate with desired value.
Given tracestate is implemented as a opaque string, the samper is responsible for ensuring the tracestate semantics are followed.

### CODE REVIEW

**Review by vishweshbankwar — APPROVED** [review]

File: src/OpenTelemetry/Trace/SamplingResult.cs [hunk 1]
```diff
@@ -58,6 +60,38 @@ public SamplingResult(SamplingDecision decision, IEnumerable<KeyValuePair<string
...
+            this.Decision = decision;
+            this.Attributes = Enumerable.Empty<KeyValuePair<string, object>>();
+            this.TraceStateString = traceStateString;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SamplingResult"/> struct.
+        /// </summary>
+        /// <param name="decision">indicates whether an activity object is recorded and sampled.</param>
+        /// <param name="attributes">Attributes associated with the sampling decision. Attributes list passed to
```
Comment (reyang) [inline src/OpenTelemetry/Trace/SamplingResult.cs:82]:
```suggestion
        /// <param name="attributes">attributes associated with the sampling decision. Attributes list passed to
```

File: src/OpenTelemetry/Trace/SamplingResult.cs [hunk 2]
```diff
@@ -70,6 +104,11 @@ public SamplingResult(SamplingDecision decision, IEnumerable<KeyValuePair<string
         /// </summary>
         public IEnumerable<KeyValuePair<string, object>> Attributes { get; }
 
+        /// <summary>
+        /// Gets the Tracestate.
```
Comment (reyang) [inline src/OpenTelemetry/Trace/SamplingResult.cs:108]:
```suggestion
        /// Gets the tracestate.
```

**Review by reyang — APPROVED** [review]
Added couple minor suggestions.
LGTM with a changelog entry.

File: src/OpenTelemetry/Trace/SamplingResult.cs [hunk 3]
```diff
@@ -42,6 +43,7 @@ public SamplingResult(bool isSampled)
         {
```
Comment (utpilla) [inline src/OpenTelemetry/Trace/SamplingResult.cs:43]: Do we not need an overload for this one?

Comment (cijothomas) [reply src/OpenTelemetry/Trace/SamplingResult.cs:43]: its optional. if there is a need, we can always add more overloads in future.

**Pull request merged**